src/TTextEdit.cpp

```diff
@@ -186,6 +196,48 @@ void TTextEdit::initDefaultSettings()
     mWrapIndentCount = 5;
 }
 
+void TTextEdit::updateHorizontalScrollBar(int lineNumber)
+{
+    if (mpConsole->getType() != TConsole::ErrorConsole) {
+        return;
+    }
+
+    if (!mIsLowerPane) {
+        // This is ONLY for the upper pane
+        if (mpConsole->mpHScrollBar && lineNumber > 1) {
+            int columnCount = getColumnCount();
+            QString& lineText = mpBuffer->lineBuffer[lineNumber];
```

Comment:
Since you are (should) not be altering the contents of the text line you should be using the `const` (read-only) `QStringList::at()` method instead of the read-write `QStringList::operator[]()` method. I.e. change this to:
```cpp
             const QString& lineText = mpBuffer->lineBuffer.at(lineNumber); 
```